services/src/main/java/org/jboss/windup/web/services/RuleDataLoader.java

```diff
@@ -134,11 +134,27 @@ private void loadRules(Configuration webConfiguration)
                         continue;
                 }
                 LOG.info("Purging existing rule data for: " + rulesPath);
-                // Delete the previous ones
-                entityManager
-                            .createNamedQuery(RuleProviderEntity.DELETE_BY_RULES_PATH)
+
+                logCurrentEntityCounts("1");
+                {
+                    // Delete the previous providers from that path
+                    List<RuleProviderEntity> providers = entityManager.createNamedQuery(RuleProviderEntity.SELECT_BY_RULES_PATH, RuleProviderEntity.class)
                             .setParameter(RuleProviderEntity.RULES_PATH_PARAM, rulesPath)
-                            .executeUpdate();
+                            .getResultList();
+                    for(RuleProviderEntity provider : providers)
+                        entityManager.remove(provider);
+                    LOG.info(RuleProviderEntity.SELECT_BY_RULES_PATH + " purged RuleProviderEntity count: " + providers.size());
+                }
+                logCurrentEntityCounts("2");
+                {
+                    // Delete the previous rule providers with no path
+                    List<RuleProviderEntity> providers = entityManager.createNamedQuery(RuleProviderEntity.SELECT_WITH_NULL_RULES_PATH, RuleProviderEntity.class)
```

Comment:
We have discussed this already. The difference is that it doesn't timeout on large sets of data.